File: bundle-deployer/src/validation/valid_base_url.ts
// Validation: Check that base_url follows expected format

import type { Bundle } from "../types/bundle.ts";

export function run(base: string, bundle: Bundle): void {
  const checkBaseUrl = 
    `https://github.com/hydrolix/integration-deployment-templates/blob/main/${base}`;
  
  if (bundle.base_url !== checkBaseUrl) {
    throw new Error(
      `Invalid ${bundle.name} base_url should be this: '${checkBaseUrl}'`
    );
  }
}

File: src/validate/valid_base_url.rs
use crate::models::bundle::Bundle;

pub async fn run(base: &str, bundle: &Bundle) -> Result<(), String> {
    let check_base_url =
        format!("https://github.com/hydrolix/integration-deployment-templates/blob/main/{base}");

    if bundle.base_url != check_base_url {
        return Err(format!(
            "ERROR: {}.{} Invalid {} base_url should be this: '{}'",
            file!(),
            line!(),
            bundle.name,
            check_base_url,
        ));
    }

    Ok(())
}
